controllers/components/kserve/kserve_support.go

```diff
@@ -220,6 +220,20 @@ func setDefaultDeploymentMode(inferenceServiceConfigMap *corev1.ConfigMap, defau
 	return nil
 }
 
+func setServiceClusterIPNone(inferenceServiceConfigMap *corev1.ConfigMap, value bool) error {
+	var serviceData map[string]interface{}
+	if err := json.Unmarshal([]byte(inferenceServiceConfigMap.Data[ServiceConfigKeyName]), &serviceData); err != nil {
```

Comment:
Would this work if `inferenceServiceConfigMap.Data[ServiceConfigKeyName]` is nil?
Upgrades will have it unset.

Reply:
the kustomize build for the component is done before this logic is executed, so for upgrades the new manifest will be applied first so the key should exist